ostree-upload-server.py

```diff
@@ -128,9 +139,33 @@ def upload(self):
                 os.close(f)
                 upload.save(real_name)
 
-                task = ReceiveTask(upload.filename, real_name,
-                                   self._repo)
+                repo_name = request.form.get('target-repo', None)
+                logging.info("Target repo: {}".format(repo_name))
+
+                if (not repo_name) and self._enforce_target_repo:
+                    logging.error("Target repo enforcement set but target repo not provided!")
+                    return self._response(400, "ERROR! target-repo parameter not set!")
```

Comment:
I think you should do all these check before you save the massive upload. I also think it should just be a requirement that the repository is specified and there's no enforce boolean.